default.nix

```diff
@@ -81,10 +81,13 @@ let
         pkg-def = stackage.${stack-pkgs.resolver};
         # The compiler referenced in the stack config
         compiler = (stack-pkgs.extras hackage).compiler or (pkg-def hackage).compiler;
+        patches = if ghcHackagePatches ? compiler.nix-name
```

Comment:
Arn't we doing a double if now?

How about:
```nix
patchesModule = ghcHackagePatches.${compiler.nix-name} or [];
```
and then use
```nix
modules = [ patchesModule ] ++ modules;
```

I would believe this works. At least `nix repl` gives me:
```
nix-repl> let foo = "hello"; in { }.${foo} or []
[ ]

nix-repl> let foo = "hello"; in { hello = [ 1 2 3 ]; }.${foo} or []
[ 1 2 3 ]
```

Reply:
Can't quite do that because that would lead to `[[]]` instead of `[]`. `or` can only be used if we want to replace it with something else, we want to replace it with nothing. I think that is going to force a double check in some senses. Pushed a prettier version though.

Reply:
Ohh! Sorry should have been a set. My bad:
```
nix-repl> let foo = "hello"; in { hello = { x = 1; }; }.${foo} or {}
{ x = 1; }

nix-repl> let foo = "hello"; in { }.${foo} or {}
{ }
```

this should allow us to get it done without resorting to `null` or am I mistaken? `ghcHackagePatches.${compiler.nix-name}` should return a set. `or {}` should return one as well. So `[ patchesModule ]` would be a list of sets `[{}]`, which I assume should work?

Reply:
I haven't looked at the consuming code to know what a spurious `{}` in the list would do. Presumably nothing. I can investigate it later today.

Reply:
I would hope it just works :-) some of the default.nix expressions in the patches/ghc* are just `{}` :-)

Reply:
Heh, in that case... pushed!